### ISSUE TITLE
Roll-up / Summary Table option

### ISSUE THREAD

**Issue by practicalparticipation** [OpenDataServices/flatten-tool#43]
In the [360 Giving Data Standard](https://threesixtygiving.github.io/standard/) we have had the concept of 'roll-up' properties. These are properties which would normally end up in a sub-table, but which, for the purpose of simple use-cases where there are only one-to-one relationships, could be included in the main table. 

For example, [this schema](https://github.com/ThreeSixtyGiving/standard/blob/master/360-giving-schema.json) generates [an Excel file](https://github.com/ThreeSixtyGiving/standard/blob/master/360-giving-schema.xslx) with only a few properties in the main tab.

However, the [360 Giving Template](https://threesixtygiving.github.io/standard/) is based on the idea of it being possible to express many of the properties from the sub-table in main. 

Often not all the properties from a referenced object/sub-table will be wanted in the summary (e.g. in summary mode we might only want the Organisation Name and Postcode to come into the main table, but not all the detailed address information or organisation locations and classifications).

To achieve this, we should:

* Create a **Summary Mode** switch for use when flattening data or generating templates
* Specify in the schema for properties of referenced objects (i.e. properties normally in sub-tables) whether they should be included in the main table when in **summary mode**. 

This would lead to the behaviour that:

* When generating a template in summary mode, many properties will be possible to record directly in the master sheet. 

* When flattening JSON in summary mode, information would be duplicated between the summary table, and sub-tables, for any property which should be included in summary mode.
 * (We need to think about how this would round-trip. I.e. would master or sub-tables take precedence if conflicts are found. I would suggest allowing sub-tables to overwrite master table values, but reporting error when there is a conflict/difference.)

**Comment by practicalparticipation** [comment]
When unflattening, elements which can be one-to-many, should come as an array.

**Comment by Bjwebb** [comment]
I've created the rollup/summary mode for use when generating the template, and when flattening JSON. I'm still working on adding a warning if there's more than one value to flatten, and the final sub bullet point (unflattening).

I've added for a flag named `--rollup` because this is the term you've used in the schema. Let me know if you'd prefer this to be e.g. `--summary` instead.

> When unflattening, elements which can be one-to-many, should come as an array.

I've added support for this. In order to remove dependency on the schema, and in keeping with other parts of the code, this field that is an array needs to be indicated in the header using `[]`.
